include/sound/sof/topology.h

```diff
@@ -56,6 +56,7 @@ struct sof_ipc_comp {
 	enum sof_comp_type type;
 	uint32_t pipeline_id;
 	uint32_t core;
+	uint8_t uuid[16];	/**< 16 Bytes UUID of the component */
```

Comment:
can you add an explanation on how you deal with older firmware and ABI issues?

Reply:
@lgirdwood proposed a new solution to avoid ABI alignment issue, will update with the new solution today.